controllers/authorino_controller.go

```diff
@@ -0,0 +1,619 @@
+					log.Error(statusErr, statusErr.Error())
+				}
+				log.Error(err, "Failed to update Authorino deployment resource", desiredDeployment.Name, desiredDeployment.Namespace)
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: time.Minute}, nil
+		}
+	}
+
+	authorinoInstance.Status.Ready = true
+	authorinoInstance.Status.LastError = ""
+	if statusErr := r.handleStatusUpdate(authorinoInstance); statusErr != nil {
+		log.Error(statusErr, statusErr.Error())
+	}
+
+	return ctrl.Result{RequeueAfter: time.Minute}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *AuthorinoReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&api.Authorino{}).
+		Complete(r)
+}
+
+func (r *AuthorinoReconciler) getAuthorinoInstance(namespacedName types.NamespacedName) (*api.Authorino, error) {
+	authorinoInstance := &api.Authorino{}
+	err := r.Get(context.TODO(), namespacedName, authorinoInstance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Authorino CR not found.")
+			return nil, nil
+		}
+		return nil, err
+	}
+	return authorinoInstance, nil
+}
+
+func (r *AuthorinoReconciler) getAuthorinoDeployment(authorino *api.Authorino) (*k8sapps.Deployment, error) {
+	authorinoDeployment := &k8sapps.Deployment{}
+	err := r.Get(context.TODO(),
+		types.NamespacedName{
+			Name:      authorino.GetName(),
+			Namespace: authorino.GetNamespace(),
+		}, authorinoDeployment)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Authorino deployment not found.")
+			return nil, nil
+		}
+		return nil, err
+	}
+	return authorinoDeployment, nil
+}
+
+func (r *AuthorinoReconciler) buildAuthorinoDeployment(authorino *api.Authorino) *k8sapps.Deployment {
+	prefix := authorino.GetName()
+
+	objectMeta := v1.ObjectMeta{
+		Name:      authorino.GetName(),
+		Namespace: authorino.GetNamespace(),
+	}
+
+	labels := labelsForAuthorino(objectMeta.GetName())
+
+	dep := &k8sapps.Deployment{
+		ObjectMeta: objectMeta,
+		Spec: k8sapps.DeploymentSpec{
+			Replicas: authorino.Spec.Replicas,
+			Selector: &v1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: k8score.PodTemplateSpec{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: k8score.PodSpec{
+					ServiceAccountName: prefix + "-authorino",
+					Containers: []k8score.Container{
+						{
+							Image:           authorino.Spec.Image,
+							ImagePullPolicy: k8score.PullPolicy(authorino.Spec.ImagePullPolicy),
+							Name:            api.AuthorinoContainerName,
+							Env:             r.buildAuthorinoEnv(authorino),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	ctrl.SetControllerReference(authorino, dep, r.Scheme)
+
+	return dep
+}
+
+func (r *AuthorinoReconciler) buildAuthorinoEnv(authorino *api.Authorino) []k8score.EnvVar {
+	envVar := []k8score.EnvVar{}
+
+	if !authorino.Spec.ClusterWide {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  "WATCH_NAMESPACE",
```

Comment:
The names of other Authorino env vars seem to be defined all in constants at the level of the API definition, which is nice, but not this one.

Reply:
I didn't include this to the list of env vars defined by Authorino because `WATCH_NAMESPACE` is defined by the `operator sdk`, therefore, authorino doesn't have much control over it but no harm in adding this env var to the list